lib/src/rewrite.rs

```diff
@@ -158,12 +158,16 @@ pub fn rebase_commit_with_options(
             }
             EmptyBehaviour::AbandonAllEmpty => *parent.tree_id() == new_tree_id,
         };
-        if should_abandon {
-            mut_repo.record_abandoned_commit(old_commit.id().clone());
-            // Record old_commit as being succeeded by the parent for the purposes of
-            // the rebase.
+        // If the user runs `jj checkout foo`, then `jj rebase -s foo -d bar`, and we
+        // drop the checked out empty commit, then the user will unknowingly
+        // have done the equivalent of `jj edit foo` instead of `jj checkout
+        // foo`. Thus, we never allow dropping the working commit. See #2766 and
```

Comment:
It would be nice to update the doc for --skip-empty in cli/commands/rebase.rs to mention that working copy commits are never abandoned.